### ISSUE TITLE
trying to figure out the documentation

### ISSUE THREAD

**Issue by daemon-byte** [react-page/react-page#723]
Hi,

It seems a lot of this documentation is out of date and it'd making it difficult to tell if this tool will do what I need it to. What I basically want is to load html into react from an external file, use this tool to edit it, and then post it out to a php script that will write it back into the html file. I thought that perhaps the HTMLRenderer component was the way to go but I can't see to get the example to work with react 16.9.

**Comment by fernandobandeira** [comment]
tl;dr this works similar to squarespace and similar website builders, you'll need to build the site basically from scratch using our tools, you can develop a script and parse your html code into our design object but that's something you'll have to do on your own.

This tool renders to HTML, but it's nothing more than a powered wysiwyg editor, it keeps the editor state as an object we call it `design` and we expect this object when editing a page, although it generates html that can be rendered in the browser and saved in the database, it'll not load any page and make it editable, if you do want to edit the code from any website and export it to something else, then you can use https://github.com/c9, if you can design the pages using our editor then this tool is for you.

As for the docs, I'll be updating all of them in the weekend and also probably launch an updated webpage with the examples and docs there.

**Comment by macrozone** [comment]
@daemon-byte yeah, the docs needs updates and more examples. The library should work with react 16.9.0 (we use it with that) and what you want is actually possible:

- the editor takes a stateobject, that is basically a json object which defines rows, cells and properties of these cells, e.g. what plugin is in the cell and what data the cell has.
- one plugin is the slate plugin which is a wysiwyg-editor and is also capable of importing html.
- e.g. in the example there is a predefined state where it imports from html: https://github.com/react-page/react-page/blob/master/examples/src/content.ts#L1603
- you could import your html that way, let the editor change it and save the output somewhere (in the same format)
- if you then need plain html you can use `ReactDOMServer.renderToString(<HTMLRenderer state={yourChangedEditorOutput} />)`
